libnmstate/netapplier.py

```diff
@@ -201,6 +212,42 @@ def _verify_change(desired_state):
     desired_state.verify_dns(current_state)
 
 
+def _prepare_vlan_ifaces_for_mtu_change(desired_state):
+    current_state = state.State(netinfo.show())
+    current_ifaces = list(six.viewvalues(current_state.interfaces))
+    desired_ifaces_states = {
+        ifstate[schema.Interface.NAME]: ifstate
+        for ifstate in desired_state[schema.Interface.KEY]
+    }
+    vlan_ifaces_states = []
+    for ifstate in current_ifaces:
+        if ifstate[schema.Interface.TYPE] != schema.InterfaceType.VLAN:
+            continue
+        ifname = ifstate[schema.Interface.NAME]
+        vlan_dstate = desired_ifaces_states.get(ifname, {})
+        if (
+            vlan_dstate.get(schema.Interface.STATE)
+            == schema.InterfaceState.ABSENT
+        ):
+            continue
+        vlan_ifaces_states.append(
+            {
```

Comment:
It might improve readability to move this to a method / function. 

Can't say I like all the 'or' stuff you're doing below, but, the only alternative that comes to mind isn't much better - updating a copy of vlan_dstate with the contents of ifstate **that are not** present in vlan_dstate.

Your call.

Reply:
Hm. Not sure about that. Would not it be too much to split that to another function just for a workaround?

Reply:
Workarounds usually have long lifespans. But that's besides the point. The real question is if the code is more efficient / readable / performs better.

I think it would be more readable the way I'm suggesting. 

But, I don't think it's enough to block - if you think this is better, you can leave it as is.

Reply:
Thanks for explaining. Doing it!

Reply:
Done!